### PR TITLE
📚 Add definition-of-done checklist for connectors into PR template

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Github doesn't support multiple PR templates. But we really need the ability to have different checklists which support different workflows. 

## How
The solution here is to put all checklists in the single PR template into collapsible sections. The expected flow is to expand the relevant checklist and use that one.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -5,11 +5,35 @@
...
+    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
+- [ ] `/test` command documented [here]() is passing. 
+    * Community members can skip this, Airbyters will run this for you. 
+- [ ] Code reviews completed
+- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
+- [ ] Documentation updated 
+    - [ ] README
+    - [ ] CHANGELOG.md
+    - [ ] Reference docs in the `docs/integrations/` directory.
+    - [ ] If adding a new connector, update the `docs/SUMMARY.md` to point to the new docs in `docs/integrations/`
```
Comment (tuliren) [inline .github/pull_request_template.md:29]: I am working on #3867 on the side, which automatically takes care of the above two entries.

**Review by tuliren — APPROVED** [review]
Nice!

File: .github/pull_request_template.md (hunk 1 above)
Comment (sherifnada) [reply .github/pull_request_template.md:29]: wonderful. I'll remove this point then.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -5,11 +5,34 @@
...
-- [ ] *Publish Docker images*
+Expand the checklist which is relevant for this PR. 
+
+<details><summary> <strong> Connector checklist </strong> </summary>
+<p>
+
+- [ ] Issue acceptance criteria met
+- [ ] Unit & integration tests added as appropriate (and are passing)
+    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
+- [ ] `/test` command documented [here]() is passing.
```
Comment (ChristopheDuong) [inline .github/pull_request_template.md:21]: Why don't we write the whole command in the checklist (with the long form so it is compatible with publish)?
`/test connector=connectors/<name>`

So we can just copy/paste/replace the name and not remember the correct syntax

**Pull request merged**